extensions/omni_txn/omni_txn.c

```diff
@@ -210,6 +218,14 @@ Datum retry(PG_FUNCTION_ARGS) {
             // go back to the context where we were
             MemoryContextSwitchTo(current_mcxt);
           }
+           // Check for timeout before retrying
+           TimestampTz current_time = GetCurrentTimestamp();
+             if (TimestampDifferenceExceeds(start_time, current_time, timeout_microsecs)) {
```

Comment:
I believe the function is not forward-declared but declared below. Would it make sense to move the function definition up?